fetch/content-length/resources/content-lengths.json

```diff
@@ -0,0 +1,122 @@
+[
+  {
+    "input": "Content-Length: 42",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 42,42",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 42\r\nContent-Length: 42",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 42\r\nContent-Length: 42,42",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 30",
+    "output": 30
+  },
+  {
+    "input": "Content-Length: 30,30",
+    "output": 30
+  },
+  {
+    "input": "Content-Length: 30\r\nContent-Length: 30",
+    "output": 30
+  },
+  {
+    "input": "Content-Length: 30\r\nContent-Length: 30,30",
+    "output": 30
+  },
+  {
+    "input": "Content-Length: 42,30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 30,42",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 42\r\nContent-Length: 30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 30\r\nContent-Length: 42",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 30,",
+    "output": null
+  },
+  {
+    "input": "Content-Length: ,30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 30\r\nContent-Length: \t",
+    "output": null
+  },
+  {
+    "input": "Content-Length: \r\nContent-Length: 30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: aaaah\r\nContent-Length: nah",
+    "output": null
+  },
+  {
+    "input": "Content-Length: aaaah, nah",
+    "output": null
+  },
+  {
+    "input": "Content-Length: aaaah\r\nContent-Length: aaaah",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: aaaah, aaaah",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: aaaah",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 42s",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 30s",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: -1",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 0x20",
+    "output": 42
+  },
+  {
+    "input": "Content-Length: 030",
+    "output": 30
+  },
+  {
+    "input": "Content-Length: 030\r\nContent-Length: 30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: 030, 30",
+    "output": null
+  },
+  {
+    "input": "Content-Length: \"30\"",
+    "output": null
+  },
+  {
+    "input": "Content-Length:30\r\nContent-Length:,\r\nContent-Length:30",
+    "output": null
+  }
```

Comment:
Tests look pretty comprehensive to me, that having been said, more suggestions:

I'm not sure we're ready to standardize on it, but "Content-Length: 50" would be nice.  Ideally, I think we should get 42 bytes and then an error, though not sure browsers (or specs?) agree with that.

Just "Content-Length: " (We have an empty value combined with a non-empty one, but no single empty value test)

We could do some extra tests around different LWS characters / multiple LWS characters, but that seems not too useful.


Thinking further ahead, we should ideally have tests with a compressed body and a Content-Length that matches the uncompressed body length (both when the compressed body is larger and smaller than the Content-Length).  Chrome considers the latter case not an error in the connection-close case, I believe, but we should probably decide on a common set of behaviors there.  It never ends.

Reply:
Is that primarily content codings or also other kinds of compression? Currently Fetch only "handles" content codings (I say "handles" as it has numerous issues), but maybe there's more to do?

Reply:
It's only Content-Encoding (Well, and however H2/H3 advertise their version of it - I'm not sure if they encode that information in a different manner, and we map it to a Content-Encoding).

Reply:
Okay, I'll update the content codings issue with your suggestion here so that can be tackled together.